packages/runtime/src/runtime.ts

```diff
@@ -363,37 +363,63 @@ export class Runtime {
     this.store.globalApps.delete(appId);
   }
 
+  // verify 'amt' microalgos can be withdrawn from account
+  assertMinBalance (amt: number, address: string): void {
+    const account = this.assertAccountDefined(this.store.accounts.get(address));
+    if ((account.amount - amt) < account.minBalance) {
+      throw new TealError(ERRORS.TEAL.INSUFFICIENT_ACCOUNT_BALANCE, {
+        amount: amt,
+        address: address
+      });
+    }
+  }
+
   // transfer ALGO as per transaction parameters
   transferAlgo (txnParam: AlgoTransferParam): void {
     const fromAccount = this.assertAccountDefined(this.store.accounts.get(txnParam.fromAccount.addr));
     const toAccount = this.assertAccountDefined(this.store.accounts.get(txnParam.toAccountAddr));
 
+    this.assertMinBalance(txnParam.amountMicroAlgos, fromAccount.address);
     fromAccount.amount -= txnParam.amountMicroAlgos; // remove 'x' algo from sender
     toAccount.amount += txnParam.amountMicroAlgos; // add 'x' algo to receiver
+
+    if (txnParam.payFlags.closeRemainderTo) {
+      const closeRemToAcc = this.assertAccountDefined(
+        this.store.accounts.get(txnParam.payFlags.closeRemainderTo));
+
+      closeRemToAcc.amount += fromAccount.amount; // transfer funds of sender to closeRemTo account
+      fromAccount.amount = 0; // close sender's account
```

Comment:
does closing means we delete account from network as well ?

Reply:
I think so. Not sure what we should do here, as we need the account in tests to check if it is closed or not.
[https://developer.algorand.org/docs/features/transactions/#close-an-account](https://developer.algorand.org/docs/features/transactions/#close-an-account)

Reply:
I think we should do what network does, then write tests for that,not the opposite

Reply:
Edit: Upon checking i don't think its removed from network. Because in crowdFunding tests, during delete application call we check the escrow balance (if it's 0 or not), after closing the escrow account. So that means account should still exist in network

Reply:
ok